packages/sources/src/api/costManagementAuthentication.js

```diff
@@ -0,0 +1,49 @@
+import { axiosInstance } from './index';
+import { COST_MANAGEMENT_API_BASE } from './constants';
+
+const delay = (interval, prms) => {
+    return new Promise(resolve => {
+        setTimeout(resolve.bind(null, prms), interval);
+    });
+}
+
+export function patchSource(data, timeout = 1000, interval = 10000) {
+    return new Promise((res, rej) => {
+        const start = Date.now();
+        const { id, ...payload } = data;
+
+        const checkSource = () => axiosInstance
+        .get(`${COST_MANAGEMENT_API_BASE}/sources/${id}/`)
+        .then(r => Promise.resolve(r))
+        .catch(error => {
+            if ((Date.now() - start) >= timeout) {
+                return Promise.reject(`Cost Management failed tracking source: ${JSON.stringify(error)}`);
+            }
+
+            return delay(interval).then(() => checkSource());
+        });
+
+        const updateSource = () => axiosInstance
```

Comment:
This request should not be pooled anymore. And you don't need to define it as a separate function, you can return this function after checkSource function.

Reply:
I would change the return of this whole function to
```jsx
return checkSource()
        .then(() => axiosinstance.patch(`${COST_MANAGEMENT_API_BASE}/sources/${id}/`, payload))
        .then(data => res(data))
        .catch(e => rej(e)) 
```